### PR TITLE
ARROW-8941: [C++/Python] Add cleanup script for arrow-nightlies conda repository

### PR DESCRIPTION
**Pull request by xhochy**

### CODE REVIEW

**Comment by xhochy** [comment]
@github-actions crossbow submit conda-cleaqn

**Comment by xhochy** [comment]
@github-actions crossbow submit conda-clean

**Comment by xhochy** [comment]
@github-actions crossbow submit conda-clean

**Comment by xhochy** [comment]
@github-actions crossbow submit conda-clean

**Comment by xhochy** [comment]
@github-actions crossbow submit conda-clean

**Comment by xhochy** [comment]
@github-actions crossbow submit conda-clean

**Comment by xhochy** [comment]
@github-actions crossbow submit conda-clean

**Comment by xhochy** [comment]
@github-actions crossbow submit conda-clean

**Comment by xhochy** [comment]
@github-actions crossbow submit conda-clean

**Comment by xhochy** [comment]
@github-actions crossbow submit conda-clean

**Comment by xhochy** [comment]
@github-actions crossbow submit conda-clean

File: dev/tasks/conda-recipes/clean.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+PLATFORMS = ["linux-64", "osx-64", "win-64"]
+PACKAGES = ["pyarrow", "arrow-cpp"]
+
+
+def packages_to_delete(package_name: str, platform: str) -> List[str]:
+    env = os.environ.copy()
+    env["CONDA_SUBDIR"] = platform
+    pkgs_json = check_output(
+        [
+            "mamba",
```
Comment (pitrou) [inline dev/tasks/conda-recipes/clean.py:21]: Is "mamba" really required for this to work?

Comment (xhochy) [reply dev/tasks/conda-recipes/clean.py:21]: It would probably also work with `conda search` with a bit higher runtime.

**Comment by xhochy** [comment]
@github-actions crossbow submit conda-clean

**Comment by xhochy** [comment]
@pitrou Successfully works with `conda` now.

**Review by kszucs — APPROVED** [review]
Wasn't aware of `anaconda remove` command. Thanks for fixing it Uwe!

**Comment by kszucs** [comment]
Created a follow-up to move this script to crossbow https://issues.apache.org/jira/browse/ARROW-9011

**Comment by kszucs** [comment]
No need to wait for the travis builds.

**Pull request closed without merging**